src/utils/webrtc/simplewebrtc/simplewebrtc.js

```diff
@@ -167,12 +167,26 @@ function SimpleWebRTC(opts) {
 
 	connection.on('stunservers', function(args) {
 		// resets/overrides the config
-		self.webrtc.config.peerConnectionConfig.iceServers = args
+		// Only items with non-empty urls may occur in iceServers according to WebRTC spec.
+		var stunServers = []
+		for (const stunServer of args) {
+			if (stunServer.urls && stunServer.urls.length) {
+				stunServers.push(stunServer)
+			}
+		}
+		self.webrtc.config.peerConnectionConfig.iceServers = stunServers
 		self.emit('stunservers', args)
 	})
 	connection.on('turnservers', function(args) {
 		// appends to the config
-		self.webrtc.config.peerConnectionConfig.iceServers = self.webrtc.config.peerConnectionConfig.iceServers.concat(args)
+		// Only items with non-empty urls may occur in iceServers according to WebRTC spec.
+		var turnServers = []
+		for (const turnServer of args) {
+			if (turnServer.urls && turnServer.urls.length) {
+				turnServers.push(turnServer)
+			}
+		}
+		self.webrtc.config.peerConnectionConfig.iceServers = self.webrtc.config.peerConnectionConfig.iceServers.concat(turnServers)
```

Comment:
This should be fixed instead in the signaling settings returned by the server, as otherwise the other clients would need to be independently fixed as well (the iOS app does not fail, although the Android app does, it seems to depend on the library used). It would be just a matter of [not appending `stun:` to the urls if `$stunServer` is empty](https://github.com/nextcloud/spreed/blob/a22a546401cfe5546afea968ce74028a2f5eb878/lib/Controller/SignalingController.php#L138) and [not appending the urls if `$stunUrls` is empty](https://github.com/nextcloud/spreed/blob/a22a546401cfe5546afea968ce74028a2f5eb878/lib/Controller/SignalingController.php#L141). Similarly, [the TURN element should not be added either if `$turnServer['server']` is empty](https://github.com/nextcloud/spreed/blob/a22a546401cfe5546afea968ce74028a2f5eb878/lib/Controller/SignalingController.php#L147-L160)

Reply:
Not experienced in nc devel - please suggest better solution on server side.